Provide correct icon size when requesting icon from the UI process

pvollan opened the pull request:
#### 91e710f11d802298346174a15f5284d3e4e94d95
<pre>
Provide correct icon size when requesting icon from the UI process
<a href="https://bugs.webkit.org/show_bug.cgi?id=242152">https://bugs.webkit.org/show_bug.cgi?id=242152</a>
&lt;rdar://90663179&gt;

Reviewed by Geoffrey Garen.

When the WebContent process is requesting icon rendering in the UI process, create the bitmap with
the &apos;iconSize&apos; constant. We use the same constant when generating bitmaps for the thumbnail icons
in the UI process.

* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::iconForAttachment):

Canonical link: <a href="https://commits.webkit.org/252066@main">https://commits.webkit.org/252066@main</a>
</pre>

geoffreygaren reviewed (approved):
So the problem case is when the thumbnail size is larger than the icon size? Is that something we could add a test for?

r=me

geoffreygaren commented on Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
```diff
@@ -337,7 +337,7 @@
     auto image = RenderThemeMac::iconForAttachment(fileName, contentType, title);
 #endif
-    return convertPlatformImageToBitmap(image.get(), size);
+    return convertPlatformImageToBitmap(image.get(), iconSize);
```
r=me

This function is pretty confusing to me. Might be worth a follow-up patch to clarify.

1. 'imageAndSize' and 'image' should be renamed to 'iconAndSize' and 'icon' since the type is IconAndSize and the data member is .icon.

2. 'size' should be part of the return value. It's super subtle that 'size' is an out parameter -- but only on one platform!

I wonder if we can test this with an API test?

pvollan replied:
> This function is pretty confusing to me. Might be worth a follow-up patch to clarify.
> 
> 1. 'imageAndSize' and 'image' should be renamed to 'iconAndSize' and 'icon' since the type is IconAndSize and the data member is .icon.
> 2. 'size' should be part of the return value. It's super subtle that 'size' is an out parameter -- but only on one platform!
> 
> I wonder if we can test this with an API test?

Yes, absolutely, I have created https://bugs.webkit.org/show_bug.cgi?id=242269 to track this.

Thanks for reviewing!

pvollan committed:
Provide correct icon size when requesting icon from the UI process
https://bugs.webkit.org/show_bug.cgi?id=242152
<rdar://90663179>

Reviewed by Geoffrey Garen.

When the WebContent process is requesting icon rendering in the UI process, create the bitmap with
the 'iconSize' constant. We use the same constant when generating bitmaps for the thumbnail icons
in the UI process.

* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::iconForAttachment):

Canonical link: https://commits.webkit.org/252066@main
